sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -784,9 +784,11 @@ static int create_sdw_codec_dai(struct device *dev,
 		if (!cpu_name)
 			return -ENOMEM;
 
-		init_dai_link(dai_links, *id, name, 1 - i, i, cpus, cpu_name,
-			      codec, link->num_adr, *id, NULL, &sdw_ops);
+		cpus[*cpus_id].dai_name = cpu_name;
```

Comment:
You need to validate cpu id since some of it's dependencies can come from topology.

Reply:
thanks, I checked the topology file. Now machine driver and topology file are implemented base on the same algorithm: visit sdw link 1 to link 4